Unable to find package $xyz with version

Sometimes I run into an issue, where `dotnet publish` fails to restore brand new NuGet package which has just been published to NuGet feed.

An example:

```log
[10:39:52.756] Job has been selected by the server ...
[10:39:52.972] Job is now building ...
[10:39:58.093] Job failed on benchmark server, stopping...
Command dotnet publish Benchmarks.csproj -c Release -o /tmp/benchmarks-server-178/psdi3gx4.wjc/benchmarks/src/Benchmarks/published /p:BenchmarksAspNetCoreVersion=5.0.0-alpha.1.20079.7 /p:MicrosoftAspNetCoreAllPackageVersion=5.0.0-alpha.1.20079.7 /p:MicrosoftAspNetCoreAppPackageVersion=5.0.0-alpha.1.20079.7 /p:BenchmarksNETStandardImplicitPackageVersion=5.0.0-alpha.1.20079.7 /p:BenchmarksNETCoreAppImplicitPackageVersion=5.0.0-alpha.1.20079.7 /p:BenchmarksRuntimeFrameworkVersion=5.0.0-alpha.1.20076.2 /p:BenchmarksTargetFramework=netcoreapp5.0 /p:MicrosoftNETCoreAppPackageVersion=5.0.0-alpha.1.20076.2 /p:MicrosoftWindowsDesktopAppPackageVersion=5.0.0-alpha.1.20079.1 /p:NETCoreAppMaximumVersion=99.9 /p:MicrosoftNETCoreApp50PackageVersion=5.0.0-alpha.1.20076.2 /p:GenerateErrorForMissingTargetingPacks=false /p:MicrosoftNETPlatformLibrary=Microsoft.NETCore.App --framework netcoreapp5.0  returned exit code 1
[STDOUT]:
Copyright (C) Microsoft Corporation. All rights reserved.

/tmp/benchmarks-server-178/psdi3gx4.wjc/benchmarks/src/Benchmarks/Benchmarks.csproj : warning NU1603: Benchmarks depends on Microsoft.AspNetCore.Mvc.NewtonsoftJson (>= 5.0.0-alpha.1.20079.7) but Microsoft.AspNetCore.Mvc.NewtonsoftJson 5.0.0-alpha.1.20079.7 was not found. An approximate best match of Microsoft.AspNetCore.Mvc.NewtonsoftJson 5.0.0-alpha1.19380.3 was resolved.
/tmp/benchmarks-server-178/psdi3gx4.wjc/benchmarks/src/Benchmarks/Benchmarks.csproj : warning NU1603: Benchmarks depends on Microsoft.AspNetCore.Server.IntegrationTesting.IIS (>= 5.0.0-alpha.1.20079.7) but Microsoft.AspNetCore.Server.IntegrationTesting.IIS 5.0.0-alpha.1.20079.7 was not found. An approximate best match of Microsoft.AspNetCore.Server.IntegrationTesting.IIS 5.0.0-alpha1.19380.3 was resolved.
/tmp/benchmarks-server-178/psdi3gx4.wjc/benchmarks/src/Benchmarks/Benchmarks.csproj : warning NU1603: Benchmarks depends on Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv (>= 5.0.0-alpha.1.20079.7) but Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv 5.0.0-alpha.1.20079.7 was not found. An approximate best match of Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv 5.0.0-alpha1.19380.3 was resolved.
/tmp/benchmarks-server-178/psdi3gx4.wjc/benchmarks/src/Benchmarks/Benchmarks.csproj : error NU1102: Unable to find package Microsoft.AspNetCore.Hosting.Abstractions with version (>= 5.0.0-alpha1.19380.3)
/tmp/benchmarks-server-178/psdi3gx4.wjc/benchmarks/src/Benchmarks/Benchmarks.csproj : error NU1102:   - Found 96 version(s) in dotnet-core [ Nearest version: 3.0.0-preview4-19121-14 ]
/tmp/benchmarks-server-178/psdi3gx4.wjc/benchmarks/src/Benchmarks/Benchmarks.csproj : error NU1102:   - Found 28 version(s) in NuGet [ Nearest version: 2.2.0 ]
/tmp/benchmarks-server-178/psdi3gx4.wjc/benchmarks/src/Benchmarks/Benchmarks.csproj : error NU1102:   - Found 11 version(s) in aspnetcore [ Nearest version: 3.0.0-preview4-19123-01 ]
/tmp/benchmarks-server-178/psdi3gx4.wjc/benchmarks/src/Benchmarks/Benchmarks.csproj : error NU1102:   - Found 0 version(s) in aspnetcore-tooling
/tmp/benchmarks-server-178/psdi3gx4.wjc/benchmarks/src/Benchmarks/Benchmarks.csproj : error NU1102:   - Found 0 version(s) in entityframeworkcore
/tmp/benchmarks-server-178/psdi3gx4.wjc/benchmarks/src/Benchmarks/Benchmarks.csproj : error NU1102:   - Found 0 version(s) in extensions
/tmp/benchmarks-server-178/psdi3gx4.wjc/benchmarks/src/Benchmarks/Benchmarks.csproj : error NU1102:   - Found 0 version(s) in temp-ignitor
/tmp/benchmarks-server-178/psdi3gx4.wjc/benchmarks/src/Benchmarks/Benchmarks.csproj : error NU1102:   - Found 0 version(s) in Npgsql-Unstable
  Restore failed in 1.98 sec for /tmp/benchmarks-server-178/psdi3gx4.wjc/benchmarks/src/Benchmarks/Benchmarks.csproj.
```

I've run into a similar issue with BenchmarkDotNet. The solution is to pass `--no-cache` to `dotnet restore`. However, I don't know how to pass `--no-cache` to `dotnet publish`

--- a/src/BenchmarksServer/Startup.cs
+++ b/src/BenchmarksServer/Startup.cs
@@ -2348,6 +2348,9 @@ namespace BenchmarkServer
                 throw new NotSupportedException($"Unsupported framework: {targetFramework}");
             }
 
+            // #1445 force no cache for restore to avoid restore failures for packages published within last 30 minutes
+            buildParameters += "/p:RestoreNoCache=true ";
+
             // Apply custom build arguments sent from the driver
             foreach (var argument in job.BuildArguments)
             {

